src/Microsoft.DotNet.Build.Tasks/GenerateCurrentVersion.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Xml;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public sealed class GenerateCurrentVersion : Task
+    {
+        /// <summary>
+        /// The path to the file that contains the date of the last comit in EPOCH time.
+        /// </summary>
+        [Required]
+        public string CurrentDate { get; set; }
+
+        /// <summary>
+        /// The Major Version that will be used for this build.
+        /// </summary>
+        [Output]
+        public string ResultCurrentVersion { get; set; }
+
+        public override bool Execute()
+        {
+            DateTime date;
+            ResultCurrentVersion = string.Empty;
+            bool isValidDate = DateTime.TryParse(CurrentDate, out date);
```

Comment:
If we are planning to do parsing here we should specific an exact date pattern given we are round tripping through a string.